From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Markus Rudy <mr@edgeless.systems>
Date: Tue, 12 May 2026 17:39:12 +0200
Subject: [PATCH] agent: don't abort in case of policy problems

Calling abort() is a rather drastic measure, as it immediately
terminates the process without cleaning up, and causes systemd to invoke
an unnecessary core dump handler instead of terminating directly. The
abrupt termination also leads to missing logs, which are crucial for
debugging the situation causing the abort.

This commit changes the failure handling to first wait, in order to give
the logger time to flush, and then to panic, which does call all the
finalizers.

Signed-off-by: Markus Rudy <mr@edgeless.systems>
---
 src/agent/src/main.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/agent/src/main.rs b/src/agent/src/main.rs
index 3cfeeb431beccb639652ca8eee9e811bf862961d..e330dea11cde7d9d7df2499a5e9a96a6784185fc 100644
--- a/src/agent/src/main.rs
+++ b/src/agent/src/main.rs
@@ -391,7 +391,7 @@ async fn start_sandbox(
         // Continuing execution without a security policy could be dangerous.
         // Give a brief moment for the logs to flush, then abort the process to stop the VM.
         tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
-        std::process::abort();
+        panic!("Failed to initialize agent policy: {:?}", e)
     }
 
     let sandbox = Arc::new(Mutex::new(s));
